File: telegram-bot-core/src/main/java/ru/tggc/telegrambotcore/handler/Handler.java
package ru.tggc.telegrambotcore.handler;

import com.pengrad.telegrambot.model.request.InlineKeyboardMarkup;
import ru.tggc.telegrambotcore.dto.PhotoDto;
import ru.tggc.telegrambotcore.dto.Response;
import ru.tggc.telegrambotcore.dto.ResponseBuilder;

import java.util.List;

public abstract class Handler {

    public Response sendSimplePhoto(PhotoDto photo) {
        return ResponseBuilder.create()
                .photo(photo)
                .build();
    }

    public Response sendSimplePhotos(List<PhotoDto> photos) {
        return ResponseBuilder.create()
                .photos(photos)
                .build();
    }

    public Response sendSimpleMessage(long chatId, String text) {
        return sendSimpleMessage(chatId, text, null);
    }

    public Response sendSimpleMessage(long chatId, String text, InlineKeyboardMarkup markup) {
        return ResponseBuilder.to(chatId)
                .message(text, markup)
                .build();
    }

    public Response sendSimpleMessages(long chatId, List<String> texts) {
        return ResponseBuilder.to(chatId)
                .messages(texts)
                .build();
    }

    public Response editMessageCaption(long chatId, Integer messageId, String caption, InlineKeyboardMarkup markup) {
        return ResponseBuilder.to(chatId)
                .edit(messageId, caption, markup)
                .build();
    }

    public Response editPhotos(Integer messageId, List<PhotoDto> photos) {
        return ResponseBuilder.create()
                .edit(photos, messageId)
                .build();
    }

    public Response editPhoto(long chatId, Integer messageId, String photoUrl, String caption) {
        return ResponseBuilder.to(chatId)
                .editPhoto(messageId, photoUrl, caption)
                .build();
    }

    public Response editSimpleMessage(long chatId, int messageId, String text) {
        return editSimpleMessage(chatId, messageId, text, null);
    }

    public Response editSimpleMessage(long chatId, int messageId, String text, InlineKeyboardMarkup markup) {
        return ResponseBuilder.to(chatId)
                .edit(messageId, text, markup)
                .build();
    }
}

File: telegram-bot-core/src/main/kotlin/ru/tggc/telegrambotcore/handler/Handler.kt
package ru.tggc.telegrambotcore.handler

import com.pengrad.telegrambot.model.request.InlineKeyboardMarkup
import ru.tggc.telegrambotcore.dto.PhotoDto
import ru.tggc.telegrambotcore.dto.Response
import ru.tggc.telegrambotcore.dto.ResponseBuilder

@Deprecated(message = "Use ctx instead")
abstract class Handler() {
    fun sendSimplePhoto(photo: PhotoDto): Response {
        return ResponseBuilder.create()
            .photo(photo)
            .build()
    }

    fun sendSimplePhotos(photos: MutableList<PhotoDto>): Response {
        return ResponseBuilder.create()
            .photos(photos)
            .build()
    }

    @JvmOverloads
    fun sendSimpleMessage(chatId: Long, text: String, markup: InlineKeyboardMarkup? = null): Response {
        return ResponseBuilder.to(chatId)
            .message(text, markup)
            .build()
    }

    fun sendSimpleMessages(chatId: Long, texts: MutableList<String>): Response {
        return ResponseBuilder.to(chatId)
            .messages(texts)
            .build()
    }

    fun editMessageCaption(chatId: Long, messageId: Int, caption: String?, markup: InlineKeyboardMarkup?): Response {
        return ResponseBuilder.to(chatId)
            .edit(messageId, caption, markup)
            .build()
    }

    fun editPhotos(messageId: Int, photos: MutableList<PhotoDto>): Response {
        return ResponseBuilder.create()
            .edit(photos, messageId)
            .build()
    }

    fun editPhoto(chatId: Long, messageId: Int, photoUrl: String?, caption: String?): Response {
        return ResponseBuilder.to(chatId)
            .editPhoto(messageId, photoUrl, caption)
            .build()
    }

    @JvmOverloads
    fun editSimpleMessage(chatId: Long, messageId: Int, text: String?, markup: InlineKeyboardMarkup? = null): Response {
        return ResponseBuilder.to(chatId)
            .edit(messageId, text, markup)
            .build()
    }
}
